### DISCUSSION TITLE
AWS S3 URL signing

### DISCUSSION THREAD

**Discussion by jamestelfer** [goreleaser/goreleaser#1782] [Q&A] [answered]
I'm publishing my release to a Homebrew tap on Github, and using the URL template to publish URLs that direct the asset download to be from S3 instead of Github. I'd like to publish signed URLs instead of making the bucket public.

I could just tell GoReleaser to not publish the formula to the target repo and post-process the formula to update the URLs. Is there a better way to do it that integrates with the tool better?

By the way, I've been down the path of using a custom download strategy for Homebrew that requires a Github token on the user's machine; the default way requires a plain text PAT though, and I'd rather avoid that.

Thanks!

**Comment by caarlos0** [comment]
I think you might be able to achieve that by changing the `brews.url_template`

https://goreleaser.com/customization/homebrew/

**Comment by jamestelfer** [comment, marked as answer]
Sorry for the long return on this question. I found that there was a limit of 7 days on signed URLs, making this idea non-workable. I ended up just publishing to a public URL even though the tap was private to avoid the bootstrap requirement on the user to set up a Personal Access Token to download from the target.
